src/Http/Http/src/Internal/RequestCookieCollection.cs

```diff
@@ -117,7 +117,7 @@ public bool ContainsKey(string key)
         return Store.ContainsKey(key);
     }
 
-    public bool TryGetValue(string key, [MaybeNullWhen(false)] out string? value)
+    public bool TryGetValue(string key, [MaybeNullWhen(false)] out string value)
```

Comment:
`value` is assigned the value `null` on line 124. You're just "hiding" the nulls from the signature, which is misleading, and isn't really fixing the original issue.

Reply:
That's what the MaybeNullWhen is for. Look at for example TryGetValue from Dictionary, and then you will see it is not nullable, but it has MaybeNullWhen. MaybeNullWhen tells the compiler that a value might be null _even if the type doesn't allow it_, but if the type is already nullable then the attribute is redundant.

You can look at the documentation of MaybeMullWhenAttribute for clarification.